### PR TITLE
add basic recording functionality

### PR DESCRIPTION
**Pull request by jlipps**
Added action recorder and code generator

### CODE REVIEW

File: app/renderer/components/Inspector/Inspector.js [hunk 1]
```diff
@@ -24,23 +29,70 @@ export default class Inspector extends Component {
...
+      {!isRecording &&
+       <Tooltip title="Start Recording">
+        <Button icon="eye-o" onClick={startRecording}/>
+       </Tooltip>
+      }
+      {isRecording &&
+       <Tooltip title="Pause Recording">
+         <Button icon="pause" onClick={pauseRecording}/>
+       </Tooltip>
+      }
```
Comment (dpgraham) [inline app/renderer/components/Inspector/Inspector.js]: Is there a way to reset the recording?

File: app/renderer/actions/Inspector.js [hunk 2]
```diff
@@ -116,10 +128,27 @@ export function unselectHoveredElement (path) {
  * Requests a method call on appium
  */
 export function applyClientMethod (params) {
-  return async (dispatch) => {
+  return async (dispatch, getState) => {
+    let isRecording = params.methodName !== 'quit' &&
+                      params.methodName !== 'source' &&
```
Comment (dpgraham) [inline app/renderer/actions/Inspector.js]: The methodName is never set to `source`. The source method is called every time.

If we were to add `source` as a client method call, though, I think we'd want to record it.

File: app/renderer/client-frameworks/framework.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+export default class Framework {
+
```
Comment (dpgraham) [inline app/renderer/client-frameworks/framework.js]: Could we put client-frameworks and highlighting directories into a utils or helpers directory?

File: app/renderer/components/Inspector/Inspector.js [hunk 4]
```diff
@@ -24,23 +29,70 @@ export default class Inspector extends Component {
...
+           size="small" type={boilerplateType}
+          />
+        </Tooltip>
+      }
+      {!!recordedActions.length &&
+        <Tooltip title="Clear Actions">
+          <Button icon="delete" onClick={clearRecording} size="small" />
+        </Tooltip>
+      }
+      {!isRecording &&
```
Comment (dpgraham) [inline app/renderer/components/Inspector/Inspector.js]: These three instances of `!!recorderActions.length &&` can be replaced with one and you can either wrap the three components in a div or make them an array. Not a big deal if left as is though.

**Comment by jlipps** [comment]
ok @dpgraham all comments addressed and ruby/python frameworks added

**Pull request merged**